### PR TITLE
boost coverage to almost 100%

### PR DESCRIPTION
**Pull request by darkdarkdragon**

### CODE REVIEW

File: src/ledger/transactions.js [hunk 1]
```diff
@@ -47,9 +47,6 @@ function parseAccountTxTransaction(tx) {
 }
 
 function counterpartyFilter(filters, tx: TransactionType) {
-  if (!filters.counterparty) {
-    return true;
```
Comment (clark800) [inline src/ledger/transactions.js]: Why is this removed?

File: test/fixtures/requests/index.js [hunk 2]
```diff
@@ -8,18 +8,30 @@ module.exports = {
   },
   prepareOrderCancellation: require('./prepare-order-cancellation'),
   preparePayment: require('./prepare-payment'),
+  preparePaymentMinAmountXRP: require('./prepare-payment-min-xrp'),
```
Comment (clark800) [inline test/fixtures/requests/index.js]: Can we use the convention of namespacing by method name (here and in other fixtures index files)? e.g. `preparePayment.minAmounXRP`

File: test/api-test.js [hunk 3]
```diff
@@ -31,11 +33,247 @@ function checkResult(expected, schemaName, response) {
...
+  });
+}
+
+
 describe('RippleAPI', function() {
   const instructions = {maxLedgerVersionOffset: 100};
   beforeEach(setupAPI.setup);
   afterEach(setupAPI.teardown);
 
+  describe('Connection', function() {
```
Comment (clark800) [inline test/api-test.js]: Maybe we can put the connection tests in a separate file?

File: src/ledger/transactions.js (hunk 1 above)
Comment (darkdarkdragon) [inline src/ledger/transactions.js]: That check is done before this function called.
https://github.com/ripple/ripple-lib/blob/develop/src/ledger/transactions.js#L79

Comment (clark800) [inline src/ledger/transactions.js]: Ok

File: test/fixtures/requests/index.js (hunk 2 above)
Comment (darkdarkdragon) [inline test/fixtures/requests/index.js]: OK

File: test/api-test.js (hunk 3 above)
Comment (darkdarkdragon) [inline test/api-test.js]: OK

File: test/fixtures/responses/index.js [hunk 4]
```diff
@@ -31,26 +34,37 @@ module.exports = {
...
+    withSettingsTx: require('./get-ledger-with-settings-tx'),
+    withStateAsHashes: require('./get-ledger-with-state-as-hashes')
   },
-  prepareOrderCancellation: require('./prepare-order-cancellation.json'),
   prepareOrder: {
     buy: require('./prepare-order.json'),
     sell: require('./prepare-order-sell.json'),
-    expiration: require('./prepare-order-expiration')
+    expiration: require('./prepare-order-expiration'),
+    cancellation: require('./prepare-order-cancellation.json'),
```
Comment (clark800) [inline test/fixtures/responses/index.js]: `prepareOrder` and `prepareOrderCancellation` should be separate namespaces because they are separate methods

File: test/fixtures/rippled/index.js [hunk 5]
```diff
@@ -17,8 +18,11 @@ module.exports = {
   },
   account_offers: require('./account-offers'),
   account_tx: require('./account-tx'),
+  account_tx_one: require('./get-transactions-one'),
   gateway_balances: require('./gateway-balances'),
   book_offers: require('./book-offers'),
+  book_offers_1: require('./book-offers-1'),
```
Comment (clark800) [inline test/fixtures/rippled/index.js]: Any more descriptive name than `1` and `2`?

File: src/transaction/trustline.js [hunk 6]
```diff
@@ -8,8 +8,7 @@ import type {Instructions, Prepare} from './types.js';
 import type {TrustLineSpecification} from '../ledger/trustlines-types.js';
 
 function convertQuality(quality) {
-  return quality === undefined ? undefined :
```
Comment (clark800) [inline src/transaction/trustline.js]: Isn't the `undefined` check necessary?

File: test/api-test.js [hunk 7]
```diff
@@ -31,11 +33,247 @@ function checkResult(expected, schemaName, response) {
...
+            assert.strictEqual(got, expect);
+            server.close();
+            done();
+          });
+        });
+
+        const options = {
+          proxy: 'ws://localhost:' + port,
+          authorization: 'authorization',
+          trustedCertificates: 'something'
```
Comment (clark800) [inline test/api-test.js]: This option is supposed to be an array of strings. What is this test case testing?

File: test/api-test.js [hunk 8]
```diff
@@ -31,11 +33,247 @@ function checkResult(expected, schemaName, response) {
...
+        };
+        const connection =
+          new utils.common.Connection(this.api.connection._url, options);
+        connection.connect().catch(done);
+        connection.connect().catch(done);
```
Comment (clark800) [inline test/api-test.js]: Is this going to call `done` twice?

File: src/transaction/trustline.js (hunk 6 above)
Comment (darkdarkdragon) [inline src/transaction/trustline.js]: It is always checked before
https://github.com/ripple/ripple-lib/blob/develop/src/transaction/trustline.js#L30-L34

Comment (clark800) [inline src/transaction/trustline.js]: Ok

File: test/api-test.js (hunk 7 above)
Comment (darkdarkdragon) [inline test/api-test.js]: It really checks nothing, just adds to coverage. And I'm not sure we need to test something here.

Comment (clark800) [inline test/api-test.js]: Hmmm, is there a way to get that coverage with a test that makes more sense?

File: test/api-test.js (hunk 8 above)
Comment (darkdarkdragon) [inline test/api-test.js]: Here `done` will be called only in case of error.
Normally it called earlier in 
`socket.on('data')` handler

File: test/api-test.js (hunk 7 above)
Comment (darkdarkdragon) [inline test/api-test.js]: I'm can't think of anything that we can test here. This param just got passed into node's `tls.connect`.

Comment (clark800) [inline test/api-test.js]: It's not going to make it to `tls.connect` if there is a validation error though right?

**Comment by clark800** [comment]
I merged this as https://github.com/ripple/ripple-lib/pull/649. Please address feedback in a separate refactoring pull request.
